entity/templates/src/main/java/package/domain/_Entity.java

```diff
@@ -108,14 +111,20 @@ public boolean equals(Object o) {
 
         <%= entityClass %> <%= entityInstance %> = (<%= entityClass %>) o;
 
-        if (id != null ? !id.equals(<%= entityInstance %>.id) : <%= entityInstance %>.id != null) return false;
+        if (id != null ? !id.equals(<%= entityInstance %>.getId()) : <%= entityInstance %>.getId() != null) return false;
 
         return true;
     }
 
     @Override
     public int hashCode() {
-        return <% if (databaseType == 'sql') { %>(int) (id ^ (id >>> 32));<% } %><% if (databaseType == 'mongodb') { %>id != null ? id.hashCode() : 0;<% } %>
+    	final int prime = 31;
```

Comment:
I've done some research because I'd got an exception in the case of alone Entity (without any inheritance by/for). The id is null but hashCode needs it be not null. http://stackoverflow.com/questions/23304219/after-adding-future-annotation-to-entity-traversableresolver-isreachable-th

Another point is for a case of inheritance. If A <= B - both have the same id. 

I think we could change this implementation of hashCode.